lib/info.go

```diff
@@ -0,0 +1,124 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package lib
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/purpleidea/mgmt/lang/funcs"
+	"github.com/purpleidea/mgmt/lang/interfaces"
+	"github.com/purpleidea/mgmt/resources"
+	"github.com/urfave/cli"
+)
+
+var myPrintln = fmt.Printf
+
+// info wraps infocmd to produce output to stdout
+func info(c *cli.Context) error {
+	output, err := infocmd(c)
+	if err != nil {
+		return err
+	}
+	w := tabwriter.NewWriter(os.Stdout, 0, 8, 2, ' ', 0)
+	fmt.Fprint(w, output.String())
+	w.Flush()
+	return nil
+}
+
+// infocmd takes the cli context and returns the requested output
+func infocmd(c *cli.Context) (bytes.Buffer, error) {
+	var out bytes.Buffer
+	var names []string
+	descriptions := make(map[string]string)
+
+	switch c.Command.Name {
+	case "resources":
+		for _, name := range resources.RegisteredResourcesNames() {
+			if name == "test" {
+				continue
+			}
+			names = append(names, name)
+
+			descriptions[name] = ""
+
+			res, err := resources.Lookup(name)
+			if err != nil {
+				continue
+			}
+
+			s := reflect.ValueOf(res).Elem()
+			typeOfT := s.Type()
+
+			var fields []string
```

Comment:
use, improve, or add to one of the functions in `resources/util.go` please,

if anything wouldn't we want the info sub commands to be separate functions?